autorest/codegen/serializers/__init__.py

```diff
@@ -178,6 +178,7 @@ def _prepare_params() -> Dict[Any, Any]:
                     for i in range(len(package_parts))
                 ],
                 "dev_status": dev_status,
+                "client_name": self.code_model.class_name,
```

Comment:
I thought we had a test for a service with token credential, guess not. Not fully necessary in this issue, but can you change one of the existing README tests to include a service with TokenCredential? Thanks! I think it's probably the most seen case of our SDKs, so would be good to hav ethat as a test

Reply:
sure, I will add a test later